### PR TITLE
Adding 'amdgpu' target device and flatbuffer for HAL executables.

### PR DESCRIPTION
**Pull request by benvanik**
The schema may change as the branch gets closer to merging but the refactoring in the compiler for serializing multiple ABIs will remain the same.

### CODE REVIEW

**Comment by benvanik** [comment]
lol didn't have the ROCM target enabled on the machine I was testing compilation on - kept wondering why the bot was finding issues I wasn't 🤦 
time for more coffee

File: runtime/src/iree/schemas/amdgpu_executable_def.fbs
Comment (ScottTodd) [inline runtime/src/iree/schemas/amdgpu_executable_def.fbs:1]: > The schema may change as the branch gets closer to merging

This PR is going into `main` right now. Did you want to stage it on a branch (https://github.com/iree-org/iree/tree/users/benvanik/amdgpu ?) if unstable? Or this is schema useful to get on `main` first and iterate while the rest of the implementation will be on the branch?

Comment (ScottTodd) [reply runtime/src/iree/schemas/amdgpu_executable_def.fbs:1]: LGTM either way, but waiting to approve because auto-merge.

Comment (benvanik) [reply runtime/src/iree/schemas/amdgpu_executable_def.fbs:1]: it's fine for now to land in main - I wanted to get the ROCMTarget changes in before linking and such. I don't expect any changes (I've written the code that uses the schema) but it won't matter (nothing is generating it).

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**